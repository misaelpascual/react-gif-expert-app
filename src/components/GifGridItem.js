import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__zoomIn animate__delay-1s">
      <img src={ url } alt={ title } />
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired
}