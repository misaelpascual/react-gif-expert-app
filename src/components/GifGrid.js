import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import { LoadingBase } from './UI/LoadingBase'

export const GifGrid = ({ category }) => {

  // const [images, setImages] = useState([]);
  const { data:images, loading } = useFetchGifs(category)


  return (
    <>
      <h3 className='animate__animated animate__fadeInDown animate__delay-1s'> { category } </h3>
      { loading && <LoadingBase /> }
      <div className="card_grid">
        {
          images?.map((img) => (
            <GifGridItem
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.protoTypes = {
  category: PropTypes.string.isRequired
}
