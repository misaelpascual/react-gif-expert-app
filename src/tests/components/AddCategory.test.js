import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en <AddCategory />', () => {

  const setCategories = jest.fn()
  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={ setCategories } />)
  })


  test('Debe de mostrarse <AddCategory /> correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe cambiar el input', () => {
    const input = wrapper.find('input')
    const value = 'Hola Mundo'

    // {} - evento
    input.simulate('change', {
      target: {
        value: value
      }
    })
  })

  test('No debe de postear la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect(setCategories).not.toHaveBeenCalled()
  })

  test('Debe de llamar el setCategories y limpiar el input', () => {
    const value = 'Prueba'

    // Simulamos el input
    wrapper.find('input').simulate('change', { target: { value: value } })

    // Simulamos el submit
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    // setCategories se debe haber llamado
    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledWith( expect.any(Array) )

    // El valor de input debe estar en ''
    expect(wrapper.find('input').prop('value')).toBe('')

  })

})