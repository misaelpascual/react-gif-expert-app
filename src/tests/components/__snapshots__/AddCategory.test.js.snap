// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrarse <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar"
    type="text"
    value=""
  />
</form>
`;
